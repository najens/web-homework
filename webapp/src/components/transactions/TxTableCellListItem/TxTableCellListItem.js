import React from 'react'
import { css } from '@emotion/core'
import { bool, string } from 'prop-types'
import { Delete, Edit } from '@material-ui/icons'

const makeDataTestId = (transactionId, fieldName) => `transaction-${transactionId}-${fieldName}`

export default function TxTableCellListItem ({ columnKey, id, isSelected, val }) {
  const isAmount = columnKey === 'amount'
  const isFiller = columnKey === 'filler'

  if (isFiller && isSelected) {
    val = (
      <div css={wrapperStyle}>
        <button className={'edit-tx-button'}>
          <Edit style={{ pointerEvents: 'none' }} />
        </button>
        <button className={'delete-tx-button'}>
          <Delete style={{ pointerEvents: 'none' }} />
        </button>
      </div>
    )
  }

  if (isAmount) {
    val = val > 0 ? `${val} $ +` : `${val * -1} $`
  }

  return (
    <td css={cellStyle(isAmount)} data-testid={makeDataTestId(id, columnKey)}>{val}</td>
  )
}

TxTableCellListItem.propTypes = {
  columnKey: string,
  id: string,
  isSelected: bool,
  val: string
}

const cellStyle = (dirRtl) => css`
  border: 1px solid #b4c1ec;
  padding: 0px 6px;
  ${dirRtl && `
  direction: rtl;
  `}
`

const wrapperStyle = css`
  align-items: center;
  display: flex;
  font-size: 8px;
  justify-content: space-between;

  & > button {
    background: none;
    border: none;
    color: #fff;
    -moz-appearance: none;
    -webkit-appearance: none;

    &:focus {
      outline: none;
    }

    &:hover {
      color: #b1d8f5;
      cursor: pointer;
    }
  }
`
