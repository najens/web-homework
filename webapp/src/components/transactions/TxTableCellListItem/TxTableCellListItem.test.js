import React from 'react'
import { create } from 'react-test-renderer'
import TxTableCellListItem from './TxTableCellListItem'

// Only testing render since this component has no business logic
describe('TxTableCellListItem component', () => {
  it('renders correctly', () => {
    const testRenderer = create(
      <TxTableCellListItem
        columnKey='amount'
        id='782a1d8d-08dc-4ef8-bab0-d922038971a2'
        val='1.00'
      />
    )
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
