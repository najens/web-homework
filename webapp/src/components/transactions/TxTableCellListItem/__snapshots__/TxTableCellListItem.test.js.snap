// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxTableCellListItem component renders correctly 1`] = `
.emotion-0 {
  border: 1px solid #b4c1ec;
  padding: 0px 6px;
  direction: rtl;
}

<td
  className="emotion-0"
  data-testid="transaction-782a1d8d-08dc-4ef8-bab0-d922038971a2-amount"
>
  1.00 $ +
</td>
`;
